lib/bridge/IrcHandler.js

```diff
@@ -573,7 +573,7 @@ IrcHandler.prototype.onMode = Promise.coroutine(function*(req, server, channel,
         }
         // Update the visibility for all rooms connected to this channel
         return this.ircBridge.publicitySyncer.updateVisibilityMap(
-            true, server.domain + ' ' + channel, enabled
+            true, server.getNetworkId() + ' ' + channel, enabled
```

Comment:
I really don't like this. I didn't say anything at the time but I'm in a less accepting mood today. I appreciate that the publicity syncer needs to key off a tuple of network ID + channel. I don't feel it is the caller's responsibility to craft the unique key. We should be calling a method which accepts a `server` and `channel` and returns the key, so we can consistently apply the concatenation rules. Otherwise, as I'm sure you now appreciate, you need to find alllll the places where we make the key and change them in the same way in order to apply a fix, which makes it really really easy to miss one place. See https://github.com/matrix-org/matrix-appservice-irc/blob/master/lib/DataStore.js#L484 for an example.